src/patch.rkt

```diff
@@ -182,27 +182,18 @@
 
   (define comb-changelists (append changelists changelists-low-locs))
   (define altns (append (^queued^) (^queuedlow^)))
-
-  (define rules-used
-    (append-map (curry map change-rule) (apply append comb-changelists)))
-  (define rule-counts
-    (for ([rgroup (group-by identity rules-used)])
-      (timeline-push! 'rules (~a (rule-name (first rgroup))) (length rgroup))))
   
   (define rewritten
     (for/fold ([done '()] #:result (reverse done))
               ([cls comb-changelists] [altn altns]
               #:when true [cl cls])
-      (let loop ([cl cl] [altn altn])
-        (cond
-          [(null? cl)
-           (cons altn done)]
-          [else
-           (define change-app (change-apply (car cl) (alt-program altn)))
-           (define prog* (apply-repr-change change-app (*context*)))
-           (if (program-body prog*)
-               (loop (cdr cl) (alt prog* (list 'change (car cl)) (list altn)))
-               done)]))))
+      (let ([cl cl] [altn altn])
+          (match-define (list subexp loc) cl)
+          (define change-app (location-do loc (alt-program altn) (const subexp)))
+          (define prog* (apply-repr-change change-app (*context*)))
+          (if (program-body prog*)
+              (cons (alt prog* (list 'change loc) (list altn)) done)
+              done))))
```

Comment:
You can remove the `let` here entirely (it defines a new variable `cl` with exactly the same value as the old variable `cl`, and likewise with `altn`.

Also, what happened to the null `cl` case?

Reply:
@bksaiki said that the null case was not needed. Will make the other change though.

Reply:
Ok, sounds good, I bet he's right.